app/assets/javascripts/controllers/article_controller.js.coffee.erb

```diff
@@ -0,0 +1,53 @@
+Blog = angular.module("Blog")
+
+
+Blog.config ($routeProvider) ->
+  $routeProvider.when('/articles/new', 
+    templateUrl: '<%= asset_path('assets/articles/new.html') %>', 
+    controller: 'ArticleController'
+  )
+  .when('/articles/:id', 
+    templateUrl: '<%= asset_path('assets/articles/show.html') %>', 
+    controller: 'ArticleController'
+  )
+  .when('/articles/:id/edit',
+    templateUrl: '<%= asset_path('assets/articles/edit.html') %>', 
+    controller: 'ArticleController'
+  )
+
+
+Blog.controller('ArticleController', ($scope, $routeParams, $location, Article, Comment) ->
+  if $routeParams.id
+    $scope.article_id = $routeParams.id
+    Article.get(id: $routeParams.id).$promise.then (article) ->
+      $scope.article = article
+      $scope.article.comments = _.map($scope.article.comments, (comment) ->
+        new Comment(comment)
+      )
+  else
+    $scope.article = new Article()
+
+  $scope.save = ($form) ->
+    return unless $form.$valid
+
+    if ($scope.article.id?)
+      promise = $scope.article.$update()
+    else
+      promise = $scope.article.$save()
+
+    promise.then (article) ->
+      $scope.errors = []
+      $location.url("/articles")
+
+    .catch (response) ->
+      $scope.handleValidationError(response, $form, $scope)
+
+  $scope.deleteComment = (comment) ->
+    comment.$delete().then (response) -> 
+      $scope.article.comments = _.reject($scope.article.comments, (comment2) -> 
```

Comment:
another option is to use ```splice``` on the array, it will do the same thing as what you have but it's a bit clearer. and the only thing you have to do is to get the index of the comment that you're removing (which you can pass in from the view https://github.com/kahlya/ruby-try-three/pull/8/files#r24347610)
```coffeescript
$scope.deleteComment = (comment, index) ->
    comment.$delete().then (response) -> 
      $scope.article.comments.splice(index, 1)
    .catch (response) ->
      alert("oh crap")
```